deployment/terraform/utils.go

```diff
@@ -39,3 +42,68 @@ func uploadBatch(sshc *ssh.Client, batch []uploadInfo) error {
 
 	return nil
 }
+
+// OpenSSHFor starts a ssh connection to the resource
+func (t *Terraform) OpenSSHFor(resource string) error {
+	output, err := t.getOutput()
+	if err != nil {
+		return fmt.Errorf("could not parse output: %w", err)
+	}
+	var cmd *exec.Cmd
+	for i, agent := range output.Agents.Value {
+		if resource == agent.Tags.Name || (i == 0 && resource == "coordinator") {
+			cmd = exec.Command("ssh", fmt.Sprintf("ubuntu@%s", agent.PublicIP))
+		}
+	}
+	for _, instance := range output.Instances.Value {
+		if resource == instance.Tags.Name {
+			cmd = exec.Command("ssh", fmt.Sprintf("ubuntu@%s", instance.PublicIP))
+		}
+	}
```

Comment:
It seems like we are unnecessarily looping through the 2 loops even after we have found an entry. Let's break if we have found our resource.